src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+using Internal.NativeFormat;
+using ILCompiler.DependencyAnalysisFramework;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Wrapper node for native layout vertex structures. This wrapper node is "abstract" as it does not 
+    /// generate any data. It is used to keep track of the dependency nodes required by a Vertex structure.
+    /// </summary>
+    internal sealed class NativeLayoutVertexNode : DependencyNodeCore<NodeFactory>
```

Comment:
We should have a discussion about this with @davidwrighton.

1. The way this is used doesn't do any interning of native layout nodes we already allocated and analyzed for dependencies (we'll just keep redoing it) every time a new `NativeLayoutVertextNode` was allocated.
2. I would prefer this was abstract and there were a bunch of children of this class that have a method to get a vertex, and know how to describe the dependencies (to improve locality and reviewability of this code). Having these two provided externally in the constructor doesn't seem like an OOP design.